lib/CodeGen/CGExprScalar.cpp

```diff
@@ -2427,6 +2427,15 @@ ScalarExprEmitter::EmitScalarPrePostIncDec(const UnaryOperator *E, LValue LV,
 
   // Next most common: pointer increment.
   } else if (const PointerType *ptr = type->getAs<PointerType>()) {
```

Comment:
According to the spec, array_ptrs of function pointers are not allowed. So I guess we do not need a check in the case for function pointers.

Reply:
Agreed - we do not need a check for the case of function pointers.